### PR TITLE
CI: pin kubectl version

### PR DESCRIPTION
**Pull request by GeorgianaElena**
Temporary fix until we figure out https://github.com/2i2c-org/infrastructure/issues/1271

### CODE REVIEW

File: .github/actions/setup-deploy/action.yaml [hunk 1]
```diff
@@ -76,6 +76,13 @@ runs:
         # behalf.
         version: v3.8.2
 
+    # Pin kubectl version to 1.23 otherwise interactions with k8s clusters versioned <=1.21 won't work.
+    # See https://github.com/2i2c-org/infrastructure/issues/1271.
+    - uses: azure/setup-kubectl@v2.0
+      with:
+        version: "1.23"
+      id: install
```
Comment (consideRatio) [inline .github/actions/setup-deploy/action.yaml:84]:
```suggestion
```

The workflow has a jobs, the jobs has steps, this step has an id set to install. This is only relevant if we want to reference the outputs of this step or similar later, which we currently don't. Due to that, let's not add an id for this step.

**Comment by consideRatio** [comment]
I don't think it closes #1271, it only mitigates the issue that makes us want to have a policy and procedure to upgrade our k8s clusters to a somewhat modern and cohesive version. I'm thinking for example that perhaps we should ensure all our clusters has version 1.21+ or similar at this point in time when 1.24 is out.

File: .github/actions/setup-deploy/action.yaml (hunk 1 above)
Comment (GeorgianaElena) [reply .github/actions/setup-deploy/action.yaml]: Oh, right! Copy-pasta gone wrong 😅 Thanks @consideRatio

**Comment by GeorgianaElena** [comment]
> I don't think it closes https://github.com/2i2c-org/infrastructure/issues/1271, it only mitigates the issue that makes us want to have a policy and procedure to upgrade our k8s clusters to a somewhat modern and cohesive version. I'm thinking for example that perhaps we should ensure all our clusters has version 1.21+ or similar at this point in time when 1.24 is out.

Good point @consideRatio! I've updated the top comment so that it won't close the issue

**Pull request merged**